src/Output.cxx

```diff
@@ -1970,7 +1971,12 @@ void ASTVisitor::OutputVarDecl(clang::VarDecl const* d, DumpNode const* dn)
   if (d->getStorageClass() == clang::SC_Extern) {
     this->OS << " extern=\"1\"";
   }
-  this->PrintMangledAttribute(d);
+
+  bool const isTranslationUnit = clang::isa<clang::TranslationUnitDecl>(d->getDeclContext());
+  if (!isTranslationUnit)
+  {
+    this->PrintMangledAttribute(d);
+  }
```

Comment:
Shouldn't this still depend on the ABI (C or C++) rather than on the namespace?  IIRC one can use `extern "C"` inside a namespace and it ignores the namespace for symbol name purposes.

Reply:
Ah good point.  Looking at just the `CXXLanguageLinkage` broke tests that shouldn't have changed.  At least the way I tried first.  I've added a failing test so I can see if I can address this test case.

Reply:
I'm stumped on how to do this actually.  The reason I am checking the namespace rather than the linkage is because as far as I can tell global variables are only mangled in a namespace.  That is:

```c++
extern "C" int c_linkage;   // C linkage, not mangled
int cpp_linkage;            // C++ linkage, not manged
namespae ns {
extern "C" int c_linkage2;  // C linkage, not manged
int cpp_linkage;            // C++ linkage manged
}
```

I verified each of these using `g++` and `nm`.

My original implementation takes these cases into account except for the last one.  I thought that explictly checking for `extern "C"` might work using `->isExternC()`:


```diff
--- a/src/Output.cxx
+++ b/src/Output.cxx
@@ -2085,8 +2085,7 @@ void ASTVisitor::OutputVarDecl(clang::VarDecl const* d, DumpNode const* dn)
   }
 
   bool const isTranslationUnit = clang::isa<clang::TranslationUnitDecl>(d->getDeclContext());
-  if (!isTranslationUnit)
-  {
+  if (!isTranslationUnit && !d->isExternC()) {
     this->PrintMangledAttribute(d);
   }
   this->PrintAttributesAttribute(d);
```

but it still fails the test in 14da45accf02d528f0e1cbe2d6a0e6954aac134c which is this code:

```c++
namespace ns {
extern "C" int start;
}
```

I've verified using `g++` and `nm` that `start` is not mangled (as you would expect), but I would also expect that the function `isExternC()` would refer to the `extern "C"` decl.  Is libclang wrong?  Or perhaps I misunderstand its intent.

`->isExternC()` or `d->getLanguageLinkage() == clang::CXXLanguageLinkage` only do the "right" (?) thing for variables if the file is C.